crates/ruff/src/rules/pycodestyle/helpers.rs

```diff
@@ -47,12 +43,11 @@ pub fn is_overlong(
                 return false;
             }
         }
+    }
 
-        // Do not enforce the line length for commented lines that end with a URL
-        // or contain only a single word.
-        if chunks.last().map_or(true, |c| URL_REGEX.is_match(c)) {
-            return false;
-        }
+    // Do not enforce the line length for lines that end with a URL.
```

Comment:
What happens if we have

```
This is a long sentence that ends with a shortened URL and, therefore, could easily be broken across multiple lines ([source](https://ruff.rs))
```

Reply:
Yeah this would be "allowed", which feels off.

I guess we could allow: (1) one word before URL (like `@see https://...`), (2) entire line is a Markdown link?

Reply:
This sounds reasonable.

Reply:
Hmm... becoming less sure of that approach. Here are some examples from Zulip that seem valid to me and are no longer flagged, but _would_ be flagged based on what I just suggested:

```py
## What portion of events are sampled (https://docs.sentry.io/platforms/javascript/configuration/sampling/):
# SENTRY_FRONTEND_SAMPLE_RATE = 1.0
# SENTRY_FRONTEND_TRACE_RATE = 0.1
```

```py
except OSError:
    print(
        FAIL + "Error: Unable to create symlinks. "
        "Make sure you have permission to create symbolic links." + ENDC
    )
    print("See this page for more information:")
    print(
        "  https://zulip.readthedocs.io/en/latest/development/setup-recommended.html#os-symlink-error"
    )
    sys.exit(1)
```

Reply:
ESLint's [max-len rule](https://eslint.org/docs/latest/rules/max-len) has many different configuration options, one to ignore comments, and another to ignore URLs. I'm not sure what the right solution is but is probably why I prefer formatters where the line-width is [a guidance](https://prettier.io/docs/en/options.html#print-width) rather than an absolute limit over line-length lint rules.

Reply:
I think we ship this for now, but... we should revise the check such that it errors if the URL _starts_ after the line-length limit.